### PR TITLE
*: implement CANCEL/DROP LOAD DATA JOB

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/insert_common.go
Comment (ekexium) [inline executor/insert_common.go:1]: Changes in this file LGTM.

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: executor/asyncloaddata/util.go [hunk 1]
```diff
@@ -178,6 +179,95 @@ func FailJob(
...
+	}
+	status := rows[0].GetEnum(0).String()
+	if status != "running" && status != "paused" {
+		return exeerrors.ErrLoadDataInvalidOperation.GenWithStackByArgs(fmt.Sprintf("need status running or paused, but got %s", status))
+	}
+	endTimeIsNull := rows[0].IsNull(1)
+	if !endTimeIsNull {
+		hasError := !rows[0].IsNull(2)
+		if hasError {
+			return exeerrors.ErrLoadDataInvalidOperation.GenWithStackByArgs("need status running or paused, but got failed")
```
Comment (D3Hunter) [inline executor/asyncloaddata/util.go:235]: make `failed` a status in job table?

File: executor/load_data.go [hunk 2]
```diff
@@ -642,10 +642,13 @@ func (e *LoadDataWorker) commitWork(ctx context.Context) (err error) {
 		tasks               uint64
 		lastScannedFileSize int64
 		currScannedFileSize int64
+		backgroundCtx       = context.Background()
```
Comment (D3Hunter) [inline executor/load_data.go:645]: err msg at line `return errors.Errorf("failed to keepalive for LOAD DATA job %d", jobID)` need updated too, job might be cancelled or deleted

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: executor/asyncloaddata/util.go (hunk 1 above)
Comment (lance6716) [reply executor/asyncloaddata/util.go:235]: currently we only have expected_status. but no actual_status, so these lines is to calculate actual_status from other columns. I plan to add actual_status later when implement PAUSE/RESUME LOAD DATA JOBS, to let PAUSE don't return until actual_status is paused :joy:

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: executor/load_data.go (hunk 2 above)
Comment (lance6716) [reply executor/load_data.go:743]: https://github.com/pingcap/tidb/pull/42137/commits/24bd2b966531ddef6b050f284c4b3bf2a873f3fa
updated

**Comment by lance6716** [comment]
ping @D3Hunter

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**